src/hooks/useBot.ts

```diff
@@ -0,0 +1,18 @@
+import { BotAPI } from './../api/bot';
+import { QueryClient, useMutation } from '@tanstack/react-query';
+
+const useSendMessage = (chatId: number) => {
+    const queryClient = new QueryClient();
+    return useMutation({
+        mutationFn: (message: string) => BotAPI.sendMessage(chatId, message),
+        onSuccess: () => {
+            queryClient.invalidateQueries({
+                queryKey: ['chat', chatId]
+            });
+        }
+    })
+}
```

Comment:
```suggestion
const queryClient = new QueryClient();
const useSendMessage = (chatId: number) => {
    return useMutation({
        mutationFn: (message: string) => BotAPI.sendMessage(chatId, message),
        onSuccess: () => {
            queryClient.invalidateQueries({
                queryKey: ['chat', chatId]
            });
        }
    })
}
```

Since queryClient creation is not a react hook and it is not part of the react lifecycle, we can probably move it outside and avoid object creation everytime when invoking this hook

Reply:
Just read through react-query source code. You can also retrieve the query client from react context with `useQueryClient`

https://github.com/TanStack/query/blob/main/packages/react-query/src/QueryClientProvider.tsx

Reply:
Yeah my bad, I was copying to my old code, but copied wrongly 🤦